src/uu/ptx/src/ptx.rs

```diff
@@ -751,8 +751,13 @@ fn write_traditional_output(
     }
 
     for word_ref in words {
-        let file_map_value: &FileContent = file_map
-            .get(&word_ref.filename)
+        let (_, file_map_value) = file_map
```

Comment:
Can you add a comment about what the purpose of this filtering is for? Its not the most intuitive, my understanding is that its to find which of the files you're referencing when there's duplicates but it would be good to explain how using those two values it lets you disambiguate

Reply:
`ptx` assigns a unique `global_line_nr` to every line read across all input files. If we run `ptx fileA fileA`, we have two entries in our file list with the same name, but covering different global line ranges (e.g., `0-10` and `11-20`).
This filter ensures that a word belonging to the second instance of the file is mapped correctly to the second file content block, rather than just picking the first file with the matching name.
I've added a comment to the code explaining this!